File: BeagleUI/Source/Layout/Padding.swift
//
//  Padding.swift
//  BeagleUI
//
//  Created by Daniel Tes on 12/09/19.
//  Copyright © 2019 Daniel Tes. All rights reserved.
//

struct Padding: Widget {
    let value: PaddingValue
    let child: Widget
}

struct PaddingValue {
    let top: UnitValue?
    let left: UnitValue?
    let right: UnitValue?
    let bottom: UnitValue?
}

File: beagle-declarative/src/main/kotlin/br/com/zup/beagle/widget/layout/Padding.kt
package br.com.zup.beagle.widget.layout

import br.com.zup.beagle.widget.core.Widget
import br.com.zup.beagle.widget.core.UnitValue

data class PaddingValue(
    val top: UnitValue? = null,
    val left: UnitValue? = null,
    val right: UnitValue? = null,
    val bottom: UnitValue? = null
)

data class Padding(
    val value: PaddingValue,
    val child: Widget
) : Widget
